packages/entity-server/src/routes/hierarchy/middleware/filter.ts

```diff
@@ -64,42 +98,37 @@ const toFilterClause = (queryClause: string) => {
   }
 
   const formattedField = isNestedField(field) ? toJsonBKey(field) : field;
-  return [formattedField, value] as [typeof formattedField, typeof value];
-};
-
-const buildCountryCodeFilter = (allowedCountries: string[], value?: string | string[]) => {
-  if (!value) {
-    return allowedCountries;
-  }
-
-  const filteredAllowedCountries = Array.isArray(value)
-    ? allowedCountries.filter(country => value.includes(country))
-    : allowedCountries.filter(country => value === country);
-
-  if (filteredAllowedCountries.length < 1) {
-    throw new Error('No access to any countries due to user permissions and country_code filter');
-  }
 
-  return filteredAllowedCountries;
+  return [formattedField, operator, value] as [
+    typeof formattedField,
+    typeof operator,
+    typeof value,
+  ];
 };
 
-export const extractFilterFromQuery = (allowedCountries: string[], queryFilter?: string) => {
+export const extractFilterFromQuery = (
+  allowedCountries: string[],
+  queryFilter?: string,
+): EntityFilter => {
   if (!queryFilter) {
     return { country_code: allowedCountries };
   }
 
   const filterClauses = queryFilter.split(CLAUSE_DELIMITER).map(toFilterClause);
   const filter: Writable<NotNullValues<EntityFilter>> = {};
 
-  filterClauses.forEach(([field, value]) => {
+  filterClauses.forEach(([field, operator, value]) => {
     const parsedValue = value.includes(MULTIPLE_VALUES_DELIMITER)
       ? value.split(MULTIPLE_VALUES_DELIMITER)
       : value;
 
-    filter[field] = parsedValue;
+    filter[field] = convertValueToAdvancedCriteria(operator, parsedValue);
```

Comment:
When I checked out this branch, it was originally showing a pretty complex typescript error here. This is the downside of dealing with complex nested generic types as we are doing here, the compiler isn't great at explaining what went wrong... However I think the real issue is that we weren't being typesafe with the `comparator` field, which the `EntityFilter` type expected to have the explicit `comparator: '!=' | 'ilike'`, whereas we were giving it `comparator: string | null`. I think if you follow all the suggestions I made here, then it should clear up the error (atleast it has in my local). Let me know if it doesn't and we can work on it together 👍

Reply:
I think that was the problem! Thanks heaps!